src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -94,10 +96,21 @@ def __init__(
         """
         self._flow_file = flow_file
         self._working_dir = Flow._resolve_working_dir(flow_file, working_dir)
-        self._flow = Flow.from_yaml(flow_file, working_dir=self._working_dir)
-        FlowValidator.ensure_flow_valid_in_batch_mode(self._flow)
+        if Path(flow_file).suffix.lower() in [".yaml", ".yml"]:
+            self._flow = Flow.from_yaml(flow_file, working_dir=self._working_dir)
+            FlowValidator.ensure_flow_valid_in_batch_mode(self._flow)
+            self._program_language = self._flow.program_language
+        # if the flow file is a python file, we need to set the flow to None and set the program language to python
```

Comment:
More explanation why we need to set flow to None? Can we just not set self._flow?

Reply:
I noticed there are several places uses `self._flow is None`, I guess maybe we can have a more accurate indicator `self._is_dag_yaml_flow`?

Reply:
We need to set the _flow to None is because we use it to check whether the current flow is a normal flow or eager flow, but `_is_dag_yaml_flow` can do the same thing, I have refined the code to use the more accurate indicator `_is_dag_yaml_flow`.